### PR TITLE
Make build flake8 clean and add flake8 precheck to .travis.yml

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
This should get us `flake8` clean and adds `flake8` to the travis config so we'll stay so. Most of the changes are straightforward, but a bit of refactoring was needed to handle some weird import order dependencies. The few `# noqa` annotations are for `__init__.py` files where unused imports are intentional.

File: confluent_kafka/kafkatest/verifiable_consumer.py [hunk 1]
```diff
@@ -131,13 +126,10 @@ def on_commit (self, err, partitions):
...
 
-
-    def do_commit (self, immediate=False, async=None):
+    def do_commit(self, immediate=False, async=None):
         """ Commit every 1000 messages or whenever there is a consume timeout
             or immediate. """
-        if (self.use_auto_commit or
-            self.consumed_msgs_at_last_commit + (0 if immediate else 1000) >
-            self.consumed_msgs):
+        if self.use_auto_commit or self.consumed_msgs_at_last_commit + (0 if immediate else 1000) > self.consumed_msgs:
```
Comment (edenhill) [inline confluent_kafka/kafkatest/verifiable_consumer.py]: I dont see how this is easier to read, but I'm not fighting the flake8ers today

File: confluent_kafka/kafkatest/verifiable_consumer.py [hunk 2]
```diff
@@ -178,11 +169,11 @@ def msg_consume (self, msg):
             return
 
         if False:
-            self.dbg('Read msg from %s [%d] @ %d' % \
+            self.dbg('Read msg from %s [%d] @ %d' %
                      (msg.topic(), msg.partition(), msg.offset()))
 
         if self.max_msgs >= 0 and self.consumed_msgs >= self.max_msgs:
-            return # ignore extra messages
+            return  # ignore extra messages
```
Comment (edenhill) [inline confluent_kafka/kafkatest/verifiable_consumer.py]: double space intentional?

File: tests/test_Consumer.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 
-from confluent_kafka import Consumer, TopicPartition, KafkaError, KafkaException, TIMESTAMP_NOT_AVAILABLE, OFFSET_INVALID, libversion
+from confluent_kafka import (Consumer, TopicPartition, KafkaError, KafkaException, TIMESTAMP_NOT_AVAILABLE,
+                             OFFSET_INVALID, libversion)
```
Comment (edenhill) [inline tests/test_Consumer.py]: ..but split over multiple lines here?
This inconsistency is at least consistent with Python's consistent inconsistency

File: confluent_kafka/kafkatest/verifiable_consumer.py (hunk 1 above)
Comment (ewencp) [inline confluent_kafka/kafkatest/verifiable_consumer.py]: It actually just didn't like the `>` comparison being split across multiple lines, I've tweaked the style a bit. flake8 doesn't complain about it, but using parentheses when they aren't necessary isn't normal python style, but here it does avoid the `\` if the condition is split across multiple lines.

File: confluent_kafka/kafkatest/verifiable_consumer.py (hunk 2 above)
Comment (ewencp) [inline confluent_kafka/kafkatest/verifiable_consumer.py]: yeah, inline comments are supposed to get 2 spaces. We can choose some warnings to disable if they get too annoying.

File: tests/test_Consumer.py (hunk 3 above)
Comment (ewencp) [inline tests/test_Consumer.py]: This one hits the line length limit, requiring it to be split over multiple lines.

**Pull request merged**